### ISSUE TITLE
Fedora 34 + WebDAV backend - not working anymore - Let's Encrypt issue?

### ISSUE THREAD

**Issue by KopfKrieg** [duplicati/duplicati#4619]
- [x] I have searched open and closed issues for duplicates.
- [x] I have searched the [forum](https://forum.duplicati.com) for related topics.

----------------------------------------

## Environment info
- **Duplicati version**: 2.0.6.3_beta
- **Operating system**: Fedora 34
- **Backend**: WebDAV (Nextcloud)

## Description
The Let's Encrypt R3 certificated expired on 2021-09-30, and now I can't make any backups. However, the certificate itself should still be valid: 
```
Certificate chain
 0 s:CN = home.kopfkrieg.org
   i:C = US, O = Let's Encrypt, CN = R3
 1 s:C = US, O = Let's Encrypt, CN = R3
   i:C = US, O = Internet Security Research Group, CN = ISRG Root X1
 2 s:C = US, O = Internet Security Research Group, CN = ISRG Root X1
   i:O = Digital Signature Trust Co., CN = DST Root CA X3
```
AFAIK only the R3 expired, ISRG Root X1 and DST Root CA X3 are still valid, so it should work™

**EDIT**: Certificates are complicated. Seems like the DST Root CA X3 expired, and the R3 was signed by both DST Root CA X3 and ISRG Root X1, and the latter is still valid (therefore the chain should be valid). https://letsencrypt.org/certificates/

Also, Duplicati (2.0.6.3_beta) on Windows 10 works fine, but not on Fedora 34 (using the same WebDAV backend for both).

## Steps to reproduce
1. Install Duplicati on Fedora 34 using the official rpm
2. Use some WebDAV storage with a certificate signed by Let's Encrypt
3. See how it fails :(

- **Actual result**:
Duplicati web ui shows: "Error: TrustFailure (Authentication failed, see inner exception.)"

- **Expected result**:
Everything should working fine.

## Debug log
```
[me:~] $ duplicati-cli backup "webdavs://home.kopfkrieg.org:443/remote.php/dav/files/<path>/?auth-username=<nope>g&auth-password=<nope>" /home/me/ --backup-name=/home/me --dbpath=/home/me/.config/Duplicati/OWPGPOMAQQ.sqlite --encryption-module=aes --compression-module=zip --dblock-size=50mb --passphrase=<nope> --retention-policy="1W:1D,4W:1W,12M:1M" --zip-compression-level=1 --zip-compression-method=Deflate --exclude-files-attributes="hidden,temporary,system" --disable-module=console-password-input --exclude="{DefaultExcludes}"
Backup started at 01.10.2021 22:10:36
Checking remote backup ...
  Listing remote folder ...
  Listing remote folder ...
  Listing remote folder ...
  Listing remote folder ...
  Listing remote folder ...
Fatal error => Error: TrustFailure (Authentication failed, see inner exception.)

System.Net.WebException: Error: TrustFailure (Authentication failed, see inner exception.) ---> System.Security.Authentication.AuthenticationException: Authentication failed, see inner exception. ---> Mono.Btls.MonoBtlsException: Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED
  at /builddir/build/BUILD/mono-6.12.0.122/external/boringssl/ssl/handshake_client.c:1132
  at Mono.Btls.MonoBtlsContext.ProcessHandshake () [0x00064] in <3f672a941b5f466fb5dc321b3015c863>:0 
  at Mono.Net.Security.MobileAuthenticatedStream.ProcessHandshake (Mono.Net.Security.AsyncOperationStatus status, System.Boolean renegotiate) [0x00106] in <3f672a941b5f466fb5dc321b3015c863>:0 
  at (wrapper remoting-invoke-with-check) Mono.Net.Security.MobileAuthenticatedStream.ProcessHandshake(Mono.Net.Security.AsyncOperationStatus,bool)
  at Mono.Net.Security.AsyncHandshakeRequest.Run (Mono.Net.Security.AsyncOperationStatus status) [0x00006] in <3f672a941b5f466fb5dc321b3015c863>:0 
  at Mono.Net.Security.AsyncProtocolRequest.ProcessOperation (System.Threading.CancellationToken cancellationToken) [0x0012a] in <3f672a941b5f466fb5dc321b3015c863>:0 
   --- End of inner exception stack trace ---
  at Mono.Net.Security.MobileAuthenticatedStream.ProcessAuthentication (System.Boolean runSynchronously, Mono.Net.Security.MonoSslAuthenticationOptions options, System.Threading.CancellationToken cancellationToken) [0x00346] in <3f672a941b5f466fb5dc321b3015c863>:0 
  at Mono.Net.Security.MonoTlsStream.CreateStream (System.Net.WebConnectionTunnel tunnel, System.Threading.CancellationToken cancellationToken) [0x001f4] in <3f672a941b5f466fb5dc321b3015c863>:0 
  at System.Net.WebConnection.CreateStream (System.Net.WebOperation operation, System.Boolean reused, System.Threading.CancellationToken cancellationToken) [0x001f5] in <3f672a941b5f466fb5dc321b3015c863>:0 
   --- End of inner exception stack trace ---
  at System.Net.WebConnection.CreateStream (System.Net.WebOperation operation, System.Boolean reused, System.Threading.CancellationToken cancellationToken) [0x00275] in <3f672a941b5f466fb5dc321b3015c863>:0 
  at System.Net.WebConnection.InitConnection (System.Net.WebOperation operation, System.Threading.CancellationToken cancellationToken) [0x0015b] in <3f672a941b5f466fb5dc321b3015c863>:0 
  at System.Net.WebOperation.Run () [0x000b7] in <3f672a941b5f466fb5dc321b3015c863>:0 
  at System.Net.WebCompletionSource`1[T].WaitForCompletion () [0x000b1] in <3f672a941b5f466fb5dc321b3015c863>:0 
  at System.Net.HttpWebRequest.RunWithTimeoutWorker[T] (System.Threading.Tasks.Task`1[TResult] workerTask, System.Int32 timeout, System.Action abort, System.Func`1[TResult] aborted, System.Threading.CancellationTokenSource cts) [0x00118] in <3f672a941b5f466fb5dc321b3015c863>:0 
  at Duplicati.Library.Main.BackendManager.List () [0x00049] in <e60bc008dd1b454d861cfacbdd3760b9>:0 
  at Duplicati.Library.Main.Operation.FilelistProcessor.RemoteListAnalysis (Duplicati.Library.Main.BackendManager backend, Duplicati.Library.Main.Options options, Duplicati.Library.Main.Database.LocalDatabase database, Duplicati.Library.Main.IBackendWriter log, System.Collections.Generic.IEnumerable`1[T] protectedFiles) [0x0000d] in <e60bc008dd1b454d861cfacbdd3760b9>:0 
  at Duplicati.Library.Main.Operation.FilelistProcessor.VerifyRemoteList (Duplicati.Library.Main.BackendManager backend, Duplicati.Library.Main.Options options, Duplicati.Library.Main.Database.LocalDatabase database, Duplicati.Library.Main.IBackendWriter log, System.Collections.Generic.IEnumerable`1[T] protectedFiles) [0x00000] in <e60bc008dd1b454d861cfacbdd3760b9>:0 
  at Duplicati.Library.Main.Operation.BackupHandler.PreBackupVerify (Duplicati.Library.Main.BackendManager backend, System.String protectedfile) [0x0011d] in <e60bc008dd1b454d861cfacbdd3760b9>:0 
  at Duplicati.Library.Main.Operation.BackupHandler.RunAsync (System.String[] sources, Duplicati.Library.Utility.IFilter filter, System.Threading.CancellationToken token) [0x01048] in <e60bc008dd1b454d861cfacbdd3760b9>:0 
  at CoCoL.ChannelExtensions.WaitForTaskOrThrow (System.Threading.Tasks.Task task) [0x00050] in <9a758ff4db6c48d6b3d4d0e5c2adf6d1>:0 
  at Duplicati.Library.Main.Operation.BackupHandler.Run (System.String[] sources, Duplicati.Library.Utility.IFilter filter, System.Threading.CancellationToken token) [0x00009] in <e60bc008dd1b454d861cfacbdd3760b9>:0 
  at Duplicati.Library.Main.Controller+<>c__DisplayClass14_0.<Backup>b__0 (Duplicati.Library.Main.BackupResults result) [0x0004b] in <e60bc008dd1b454d861cfacbdd3760b9>:0 
  at Duplicati.Library.Main.Controller.RunAction[T] (T result, System.String[]& paths, Duplicati.Library.Utility.IFilter& filter, System.Action`1[T] method) [0x0026f] in <e60bc008dd1b454d861cfacbdd3760b9>:0 
  at Duplicati.Library.Main.Controller.Backup (System.String[] inputsources, Duplicati.Library.Utility.IFilter filter) [0x00074] in <e60bc008dd1b454d861cfacbdd3760b9>:0 
  at Duplicati.CommandLine.Commands.Backup (System.IO.TextWriter outwriter, System.Action`1[T] setup, System.Collections.Generic.List`1[T] args, System.Collections.Generic.Dictionary`2[TKey,TValue] options, Duplicati.Library.Utility.IFilter filter) [0x00119] in <eb902dee3a2c4c3fbc8192580b4a42b2>:0 
  at (wrapper delegate-invoke) System.Func`6[System.IO.TextWriter,System.Action`1[Duplicati.Library.Main.Controller],System.Collections.Generic.List`1[System.String],System.Collections.Generic.Dictionary`2[System.String,System.String],Duplicati.Library.Utility.IFilter,System.Int32].invoke_TResult_T1_T2_T3_T4_T5(System.IO.TextWriter,System.Action`1<Duplicati.Library.Main.Controller>,System.Collections.Generic.List`1<string>,System.Collections.Generic.Dictionary`2<string, string>,Duplicati.Library.Utility.IFilter)
  at Duplicati.CommandLine.Program.ParseCommandLine (System.IO.TextWriter outwriter, System.Action`1[T] setup, System.Boolean& verboseErrors, System.String[] args) [0x00342] in <eb902dee3a2c4c3fbc8192580b4a42b2>:0 
  at Duplicati.CommandLine.Program.RunCommandLine (System.IO.TextWriter outwriter, System.IO.TextWriter errwriter, System.Action`1[T] setup, System.String[] args) [0x00002] in <eb902dee3a2c4c3fbc8192580b4a42b2>:0 
[me:~] $ 
```

**Comment by ts678** [comment]
FYI [Http send report errors / duplicati-monitoring](https://forum.duplicati.com/t/http-send-report-errors-duplicati-monitoring/13157) is discussing this, has links to other references, and is gathering workarounds.
There's an [RHEL/CentOS 7 Fix for Let’s Encrypt Change](https://forum.duplicati.com/t/http-send-report-errors-duplicati-monitoring/13157/23?u=ts678}) article cited there that might work on Fedora 34 as well if they're similar.
[Failed to connect: Error: TrustFailure #3535](https://github.com/duplicati/duplicati/issues/3535#issuecomment-932727296) is an old issue that got brought into the Let's Encrypt problem, and got into Docker.

**Comment by ts678** [comment]
[Bug 1733560](https://bugzilla.mozilla.org/show_bug.cgi?id=1733560) - Remove Expired Root Certificates from NSS - DST Root CA X3. was fixed at Mozilla 13 Dec 2021, maybe NSS 3.74.
Fedora rpms / ca-certificates commit history suggests it's in Fedora 36. Regardless, I don't think there's anything more to do here.
Last call for feedback, otherwise this issue will eventually close.
